dao/src/main/java/com/iluwatar/dao/Customer.java

```diff
@@ -37,34 +37,39 @@ public String getLastName() {
         return lastName;
     }
 
-    public void setLastName(String lastName) {
+    public void setLastName(final String lastName) {
         this.lastName = lastName;
     }
 
     @Override
     public String toString() {
         return "Customer{" +
-                "id=" + id +
-                ", firstName='" + firstName + '\'' +
-                ", lastName='" + lastName + '\'' +
+                "id=" + getId() +
+                ", firstName='" + getFirstName() + '\'' +
+                ", lastName='" + getLastName() + '\'' +
                 '}';
     }
 
+    /**
+     * Checks if two objects are the same.
+     * 
+     * @return true if the two objects are Customer objects and have the same id value, false otherwise.
+     */
     @Override
-    public boolean equals(Object o) {
-        if (this == o) return true;
-        if (o == null || getClass() != o.getClass()) return false;
-
-        Customer customer = (Customer) o;
-
-        if (id != customer.id) return false;
-
-        return true;
+    public boolean equals(final Object o) {
+    	boolean isEqual = false;
+    	final Customer customer = (Customer) o;
```

Comment:
Logic: We should not promote incorrect conventions. There should not be a blind cast to `Customer` in `equals()`.
```java
public boolean equals(Object o) {
  if (o == this) { 
    return true; 
  }
  if (!(o instanceof Customer)) {
    return false;
  }
  // Its safe to cast here. Then the usual equality checks
}